src/TEdit/ViewModel/WorldViewModel.Editor.cs

```diff
@@ -54,6 +56,109 @@ public void EditCopy()
             _clipboard.LoadedBuffers.Add(_clipboard.Buffer);
         }
 
+        public void CropWorld()
+        {
+            if (CurrentWorld == null) return;
+            if (!CanCopy())
+                return;
+
+            bool addBounderies = false;
+
+            if (MessageBox.Show(
+                "This will generate a new world with a selected region. Any progress done to this world will be lost, Continue?",
+                "Crop World?",
+                MessageBoxButton.YesNo,
+                MessageBoxImage.Question,
+                MessageBoxResult.Yes) != MessageBoxResult.Yes)
+                return;
+
+            if (MessageBox.Show(
+                "Add \"edge of world\" boundaries?",
+                "Crop World:",
+                MessageBoxButton.YesNo,
+                MessageBoxImage.Question,
+                MessageBoxResult.Yes) == MessageBoxResult.Yes)
+            { addBounderies = true; };
+
+            // Create clipboard
+            _clipboard.Buffer = _clipboard.GetSelectionBuffer();
+            _clipboard.LoadedBuffers.Add(_clipboard.Buffer);
+
+            // Generate New Worlds
+            _loadTimer.Reset();
+            _loadTimer.Start();
+            _saveTimer.Stop();
+
+            World w = new World();
+            Task.Factory.StartNew(() =>
+            {
+                // 41 block buffer left and top for "off-screen" on all sides.
+                // 42 block buffer right and bottom for "off-screen" on all sides. 
+                if (addBounderies)
+                { w = new World(_clipboard.Buffer.Size.Y + 83, _clipboard.Buffer.Size.X + 83, "TEdit World"); }
+                else
+                { w = new World(_clipboard.Buffer.Size.Y, _clipboard.Buffer.Size.X, "TEdit World"); }
+                
+                w.Version = World.CompatibleVersion;
+                w.GroundLevel = 350;
+                w.RockLevel = 480;
+                w.ResetTime();
+                w.CreationTime = System.DateTime.Now.ToBinary();
+
+                w.Seed = (new Random()).Next(0, int.MaxValue).ToString();
+                w.SpawnX = (int)(w.TilesWide / 2);
+                w.SpawnY = (int)Math.Max(0, w.GroundLevel / 2);
+                w.GroundLevel = (int)w.GroundLevel;
+                w.RockLevel = (int)w.RockLevel;

```

Comment:
@BinaryConstruct look into shifting this to preserve ground level when cropping height